fdbserver/storageserver.actor.cpp

```diff
@@ -3962,7 +3962,7 @@ ACTOR Future<GetRangeReqAndResultRef> quickGetKeyValues(
 		// TODO: is DefaultPromiseEndpoint the best priority for this?
 		tr.trState->taskID = TaskPriority::DefaultPromiseEndpoint;
 		Future<RangeResult> rangeResultFuture =
-		    tr.getRange(prefixRange(prefix), GetRangeLimits::ROW_LIMIT_UNLIMITED, Snapshot::True);
+		    tr.getRange(prefixRange(getRange.begin.getKey()), GetRangeLimits::ROW_LIMIT_UNLIMITED, Snapshot::True);
```

Comment:
I think the problem is not `prefix` is out-of-scope. It's the temporary object from `prefixRange` that became invalid in the next line's `wait()` call. We need to ensure this object is valid. So I'd propose this change:
```
auto range = prefixRange(prefix);
Future<RangeResult> rangeResultFuture =
 		    tr.getRange(range, GetRangeLimits::ROW_LIMIT_UNLIMITED, Snapshot::True);
RangeResult rangeResult = holdWhile(range, wait(rangeResultFuture));
```

Reply:
> It's the temporary object from prefixRange

tr.getRange takes a standalone, so it takes shared ownership of `range`. `range` doesn't need to outlive the returned future.

Reply:
> > It's the temporary object from prefixRange
> 
> tr.getRange takes a standalone, so it takes shared ownership of `range`. `range` doesn't need to outlive the returned future.

Right! I've missed this.

Maybe we just need to change the type of `prefix` in `quickGetKeyValues()` from `StringRef` to `Key` to fix this. I'm still confused as whey `prefix`'s memory becomes invalid, since the caller always passes a `Key mappedKey` in.